packages/dds/tree/src/test/tableSchema.spec.ts

```diff
@@ -1455,16 +1458,159 @@ describe("TableFactory unit tests", () => {
 		assert.equal(column, column0);
 	});
 
+	// The following tests demonstrate more complex usage scenarios.
+	describe("User stories", () => {
+		// The following test is a demonstration of how users can add runtime type constraints for cells in a column.
+		// TODO: in the future, we want to find better ways to express such type constraints directly through the factory APIs.
+		it("Using column props for runtime type constraints", () => {
+			// The underlying storage format for the cell data is a string.
+			const Cell = schemaFactory.string;
+			type Cell = TreeNodeFromImplicitAllowedTypes<typeof Cell>;
+
+			enum ColumnType {
+				/** A raw string */
+				String = "string",
+				/** An integer - must be parsable by `Number.parseInt` */
+				Integer = "integer",
+				/** A date - must be parsable by `Date.parse` */
+				Date = "date",
+			}
+
+			const ColumnTypeNodes = adaptEnum(schemaFactory, ColumnType);
+			// Defined the types of the nodes which correspond to this the schema.
+			type ColumnTypeNodes = TreeNodeFromImplicitAllowedTypes<typeof ColumnTypeNodes.schema>;
+
+			class MyColumn extends TableSchema.column({
+				schemaFactory,
+				cell: Cell,
+				props: schemaFactory.object("TableColumnProps", {
+					/** Column label */
+					label: schemaFactory.string,
+					/** ColumnType */
+					type: ColumnTypeNodes.schema,
+				}),
+			}) {
+				/**
+				 * Checks if the provided cell value is valid for the column type.
+				 * If invalid, the UI will display the cell contents highlighted in red.
```

Comment:
I imagine this is referring to some theoretical UI? Might be better to make that clear that this code doesn't somehow magically color things and the relevant code doesn't exist.
```suggestion
				 * This could be used by a view of this table to highlight invalid cells.
```

Reply:
Theoretical, correct.

Reply:
Example removed. Will follow up separately.